client/packages/core/src/index.ts

```diff
@@ -232,13 +228,16 @@ type LifecycleSubscriptionState<
   isLoading: boolean;
 };
 
-type InstaQLLifecycleState<
-  Schema,
-  Q,
-  UseDates extends boolean | undefined = false,
-> = InstaQLSubscriptionState<Schema, Q, UseDates> & {
-  isLoading: boolean;
-};
+type InstaQLLifecycleState<Schema, Q, UseDates extends boolean = false> =
+  | (InstaQLSubscriptionState<Schema, Q, UseDates> & {
+      isLoading: boolean;
+    })
+  | {
```

Comment:
We were missing this case from the lifecycle states. It was either `data` and no `error` or `error` and no `data`, but not both no `data` and no `error`.